libr/debug/core.c

```diff
@@ -0,0 +1,134 @@
+/* radare - LGPL - Copyright 2009-2015 - kur0 */
+
+#include <r_debug.h>
+
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <signal.h>
+
+#include <sys/proc.h>
+#include <sys/sysctl.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+
+#include <mach/mach.h>
+#include <mach/mach_vm.h>
+#include <mach/machine.h>
+#include <mach/thread_status.h>
+#include <mach/vm_region.h>
+
+#include <mach-o/loader.h>
+
+// TODO: Put this code in an if that checks if the target is a mach kernel.
```

Comment:
this is mach0 specific, must be implemented in a plugin, or at least in a separate file debug/mach0 or so